### PR TITLE
Refactor out 'filter'

### PR DESCRIPTION
**Pull request by zdrve**
Also one bug fix. See its own commit.

### CODE REVIEW

**Comment by zdrve** [comment]
I intend to do a similar job on 'generator' and 'saver' in forthcoming PRs.

File: lib/kennel.rb [hunk 1]
```diff
@@ -127,10 +132,10 @@ def generated
           load_all
 
           projects = Models::Project.recursive_subclasses.map(&:new)
-          filter_resources!(projects, :kennel_id, project_filter, "projects", "PROJECT")
+          Kennel::Filter.filter_resources!(projects, :kennel_id, filter.project_filter, "projects", "PROJECT")
```
Comment (grosser) [inline lib/kennel.rb:135]: might make sense to:
```suggestion
          filter.filter_projects!(projects)
```
so the filter can hold have all the "PROJECT" logic

Comment (grosser) [reply lib/kennel.rb:135]: working on that ...

Comment (grosser) [reply lib/kennel.rb:135]: https://github.com/grosser/kennel/pull/192

**Pull request merged**